src/backend/utils/adt/agtype.c

```diff
@@ -3871,7 +3870,7 @@ Datum agtype_hash_cmp(PG_FUNCTION_ARGS)
     uint64 seed = 0xF0F0F0F0;
 
     if (PG_ARGISNULL(0))
-        PG_RETURN_INT16(0);
+        PG_RETURN_INT64(0);
```

Comment:
From postgres documentation[1] hash indexes are size of int32.

This should be changed to `PG_RETURN_INT32`.  The same tests are passing but there is a good chance we are winning on performance. 

1 : https://www.postgresql.org/docs/current/indexes-types.html#INDEXES-TYPES-HASH